vignettes/Treatment.Rmd

```diff
@@ -15,50 +15,126 @@ knitr::opts_chunk$set(
+
+## Functions
+
+Drug parameters can be incorporated into a complete parameter set using the `set_drugs` function which takes the full parameter set and a list of drug parameter sets.
+
+A treatment regimen for each drug can then be described using the `set_clinical_treatment()` function which takes the drug index, a vector of timesteps at which a change in coverage occurs (where the initial coverage is 0 until the first timestep specified) and a vector of coverages for the drug that correspond with the timestep changes, as well as the complete parameter set.
+
+Multiple drugs can be modelled simultaneously, with treatment coverage that can be specified for each drug. This function must be used for each drug included (e.g., if there are two drugs, `set_clinical_treatment()` must be called twice to specify the treatment plan).
+
+
+# Example
+
+We will run a simulation for one year using AL and DHA-PQP treatment regimens. We begin our AL treatment on day 100 at 40% coverage and provide this treatment for 200 days before the treatment stops. DHA-PQP treatment also has 40% coverage, but begins on day 200 and lasts for 100 days, also ending on day 300. Note that the sum of treatment coverages cannot exceed 100%.
+
+Prior to the simulation, the function `set_equilibrium` must also be used to generate equilibrium values for the human and mosquito populations.
+
+## Parameterisation and simulation
 
 ```{r}
+# Daily simulation timesteps for two years
 year <- 365
-sim_length <- 2 * year
+sim_length <- 1 * year
+
+# With a population size of 1000
 human_population <- 1000
 
-simparams <- get_parameters(list(human_population = human_population))
-simparams <- set_drugs(simparams, list(AL_params, DHA_PQP_params))
-```
+# Set human population size
+simparams <- get_parameters(overrides = list(human_population = human_population))
 
-Run the model for some starting EIRs and fts:
+# Update parameter set with chosen drug-specific parameters (AL and DHA/PQP)
+drug_params <- set_drugs(simparams, list(AL_params, DHA_PQP_params))
 
-```{r}
-outputs <- list()
-starting_EIR <- 10
-fts <- c(0, .25, .5, .75, 1.)
-
-for (ft in fts) {
-  simparams <- set_clinical_treatment(simparams, 1, 1, ft / 2)
-  simparams <- set_clinical_treatment(simparams, 2, 1, ft / 2)
-  simparams <- set_equilibrium(simparams, starting_EIR)
-  output <- run_simulation(sim_length, simparams)
-  # cut out the warmup
-  output <- output[output$timestep > year,]
-  outputs[[length(outputs) + 1]] <- output
-}
+# Choose initial EIR to 10
+starting_EIR <- 10 
+
+# Set treatment program for AL (drug index = 1)
+treatment_params <- set_clinical_treatment(
+  parameters = drug_params,
+  drug = 1,
+  timesteps =  c(100,300), # Treatment coverage changes on day 100 and day 200
```

Comment:
Make code and comment consistent with text above which says 40% coverage starting at day 100 for 200 days ( = days 100-300).